evalml/pipelines/components/utils.py

```diff
@@ -103,3 +108,54 @@ def handle_component_class(component_class):
         raise MissingComponentError('Component "{}" was not found'.format(component_class))
     component_class = component_classes[component_class]
     return component_class
+
+
+class WrappedSKClassifier(BaseEstimator, ClassifierMixin):
+    def __init__(self, pipeline):
+        self.pipeline = pipeline
+
+    def fit(self, X, y):
+        X, y = check_X_y(X, y)
+        self.classes_ = unique_labels(y)
+        self.X_ = X
+        self.y_ = y
+        self.is_fitted_ = True
+        self.pipeline.fit(X, y)
+        return self
+
+    def predict(self, X):
+        X = check_array(X)
+        check_is_fitted(self, 'is_fitted_')
+        return self.pipeline.predict(X).to_numpy()
+
+    def predict_proba(self, X):
+        return self.pipeline.predict_proba(X).to_numpy()
+
+
+class WrappedSKRegressor(BaseEstimator, RegressorMixin):
+    def __init__(self, pipeline):
+        self.pipeline = pipeline
+
+    def fit(self, X, y):
+        X, y = check_X_y(X, y)
+        self.pipeline.fit(X, y)
+        return self
+
+    def predict(self, X):
+        return self.pipeline.predict(X).to_numpy()
+
+
+def scikit_learn_wrapped_estimator(evalml_obj, is_pipeline=True):
+    """Wrap an EvalML pipeline or estimator in a scikit-learn estimator."""
+    if is_pipeline:
+        if evalml_obj.problem_type == ProblemTypes.REGRESSION:
+            return WrappedSKRegressor(evalml_obj)
+        elif evalml_obj.problem_type == ProblemTypes.BINARY or evalml_obj.problem_type == ProblemTypes.MULTICLASS:
+            return WrappedSKClassifier(evalml_obj)
+    else:
+        # EvalML Estimator
+        if evalml_obj.supported_problem_types == [ProblemTypes.REGRESSION]:
+            return WrappedSKRegressor(evalml_obj)
+        elif evalml_obj.supported_problem_types == [ProblemTypes.BINARY, ProblemTypes.MULTICLASS]:
+            return WrappedSKClassifier(evalml_obj)
+    raise ValueError("Could not wrap EvalML object in scikit-learn wrapper.")
```

Comment:
Two notes:
1. You can replace `is_pipeline` with `isinstance(evalml_obj, PipelineBase)`
2. This code is great but augh we're still being chased by this issue of estimators not having an explicit problem type!

Reply:
Hmm, doing so will cause a circular dependency (having to import `PipelineBase`) but I do like this better. For now, will put the import statement in the method. (

Reply:
Ah good point on the circular dep. Now that you mention it, since this is supposed to work for pipelines and components, it should go in `evalml/pipelines/utils.py`. Then the circular dep wouldn't happen either :)

Reply:
I don't think you updated the impl but that's ok!

Reply:
@dsherry Oh, I believe the circular dependency still happens if we put it in `evalml/pipelines/utils.py` since the method is being used in a component (`StackedEnsembleBase`). I think this is another case where we should think about separating pipelines from components to avoid the problem!

Also, I currently see on main that `from evalml.pipelines.pipeline_base import PipelineBase` is called in `scikit_learn_wrapped_estimator`. Is that different from what you see? :o

Reply:
Ah, interesting, right... this component now knows about pipelines. Ok, got it, thanks.

Oh lol I was looking for it in this PR. Got it.

Reply:
Yup! Definitely a bit odd. This works for now but could be indicative of some API considerations we'll need to make.

And cool, glad that everything was in place! Odd that Github doesn't mark it as outdated but I guess the change was not exactly to this block 😂